### PR TITLE
Fix Typo, Make code more concise, Bump verions

### PR DESCRIPTION
**Pull request by ligi**

### CODE REVIEW

File: sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt [hunk 1]
```diff
@@ -199,65 +129,45 @@ object Sha512 {
     private fun padMessage(input: ByteArray): ByteArray {
         // Need to append at least 17 bytes (16 for length of the message, and 1 for the '1' bit)
         // then fill with 0s until multiple of 128 bytes
-        var size = input.size + 17
-
-        val remainder = size % 128
-        if (remainder != 0) {
-            size += 128 - remainder
+        val size = (input.size + 17).let {
+            if (it % 128 == 0) it else it + 128 - it % 128
```
Comment (friedger) [inline sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt]: I thought I could save one division here.

File: sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt [hunk 2]
```diff
@@ -199,65 +129,45 @@ object Sha512 {
...
+        LongArray(16) { j -> arrToLong(input, it * 128 + j * 8) }
     }
 
     // Calculates the expanded message blocks W0-W79
-    private fun calculateMessageBlocks(M: Array<LongArray>): Array<LongArray> {
-        val W = Array(M.size) { LongArray(80) }
-        // For each block in the input
+    private fun calculateMessageBlocks(M: Array<LongArray>) = Array(M.size) { i ->
+        LongArray(80) { j -> if (j < 16) M[i][j] else 0 }
+    }.apply {
```
Comment (friedger) [inline sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt:168]: What is the motivation to do it this way?

**Review by friedger — APPROVED** [review]
Looks good. Sometimes, the kotlin syntax is not that easy to read for me.

File: sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt (hunk 2 above)
Comment (ligi) [reply sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt:168]: It's more concise and IMHO better expressign what it is doing

File: sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt (hunk 1 above)
Comment (ligi) [reply sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt]: Yea  I strongly wanted the val here - might have wanted it too much

**Comment by ligi** [comment]
prevented the second division via 20f82ed6c2cbf1e06cd45e3a6477b646f8172fa0

**Pull request merged**